logfield.go

```diff
@@ -0,0 +1,136 @@
+package opentracing
```

Comment:
yeah, I started down that road and moved it here because I thought I would eventually want `LogReference` and/or other things defined in the main OT package. Meh.